File: Exercises/T3/src/java/Main.java
/**
 * @author: Jorge Augusto Salgado Salhani - 8927418
 */



package Exercises.T3.src.java;
import Exercises.T3.src.java.Cargos.Assistente;
import Exercises.T3.src.java.Cargos.Gerente;
import Exercises.T3.src.java.Cargos.Vendedora;
import Exercises.T3.src.java.Entrada.EntradaTeclado;
import Exercises.T3.src.java.GestaoABC.FolhaSalarial;

public class Main {
  
  public static void main(String args[]) {
    String cpf = "12345678909";
    
    Gerente.verificaCPF(cpf);
    Gerente gerente = new Gerente("Iara", cpf, 10.0);

    cpf = "42092105876";
    Vendedora.verificaCPF(cpf);
    Vendedora vendedora = new Vendedora("Iara", cpf, 10.0, 2.0);

    cpf = "12345678909";
    Assistente.verificaCPF(cpf);
    Assistente assistente = new Assistente("Iara", cpf, 10.0);

    FolhaSalarial folhaSalarial = new FolhaSalarial();
    folhaSalarial.adicionarFuncionaria(assistente);
    folhaSalarial.adicionarFuncionaria(vendedora);
    folhaSalarial.adicionarFuncionaria(gerente);

    System.out.println(folhaSalarial);
  }
}

File: Exercises/T3/src/python/main.py
"""
@author: Jorge Augusto Salgado Salhani - 8927418
"""

from Cargos.Assistente import Assistente
from Cargos.Gerente import Gerente
from Cargos.Vendedora import Vendedora
from GestaoABC.FolhaSalarial import FolhaSalarial

if __name__ == '__main__':
    cpf = "12345678909"
    Gerente.verificaCPF(cpf)
    gerente = Gerente("Flora", cpf, 20.00)

    cpf = "42092105876"
    Vendedora.verificaCPF(cpf)
    vendedora = Vendedora("Flora", cpf, 20.00, 15.00)

    cpf = "42092105876"
    Assistente.verificaCPF(cpf)
    assistente = Assistente("Flora", cpf, 20.00)

    folhaSalarial = FolhaSalarial()
    folhaSalarial.adicionarFuncionaria(gerente)
    folhaSalarial.adicionarFuncionaria(vendedora)
    folhaSalarial.adicionarFuncionaria(assistente)
    print(folhaSalarial.__str__())
